src/components/LineTabPicker.tsx

```diff
@@ -35,38 +35,39 @@ export const LineTabPicker: React.FC<LineTabPickerProps> = ({ lines, trainsByRou
     }, [lineSearchParam, totalTrainCount]);
 
     return (
-        <TabList className="tab-picker" aria-label="Select a line" ref={wrapperRef}>
-            <div className="selected-indicator" ref={selectedIndicatorRef} />
-            {lines.map((line) => {
-                const trains = getTrainRoutePairsForLine(trainsByRoute, line.routes);
-                return (
-                    <Tab
-                        id={line.name}
-                        className="tab"
-                        key={line.name}
-                        as="div"
-                        data-color={line.color}
-                        onClick={() => {
-                            setLineSearchParam(line);
-                        }}
-                    >
-                        <div
-                            aria-label={line.name + ' line'}
-                            className="icon"
-                            style={{ backgroundColor: line.color }}
+        <TabProvider>
```

Comment:
What does adding `TabProvider` do, out of curiosity?

Reply:
It's required to store the state in the new version of the library (uses react hook state instead of older `state.`